packages/quantic/force-app/main/default/lwc/quanticResultsPerPage/__tests__/quanticResultsPerPage.test.js

```diff
@@ -0,0 +1,360 @@
+        selectors.componentError
+      );
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('controller initialization', () => {
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+    });
+
+    it('should build the results per page and subscribe to state', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          initialState: {
+            numberOfResults: 10,
+          },
+        }
+      );
+      expect(functionsMocks.buildSearchStatus).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildSearchStatus).toHaveBeenCalledWith(
+        exampleEngine
+      );
+    });
+  });
+
+  describe('with default options', () => {
+    it('should properly render with the default choice selected', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const resultsPerPage = element.shadowRoot.querySelector(
+        selectors.resultsPerPage
+      );
+      expect(resultsPerPage).not.toBeNull();
+
+      const resultsPerPageOptions = Array.from(
+        element.shadowRoot.querySelectorAll(selectors.resultsPerPageOption)
+      );
+      expect(resultsPerPageOptions).toHaveLength(defaultChoices.length);
+
+      resultsPerPageOptions.forEach((option, index) => {
+        expect(option.number).toEqual(defaultChoices[index]);
+        expect(option.selected).toBe(
+          defaultChoices[index] === defaultInitialChoice
+        );
+      });
+    });
+  });
+
+  describe('with a custom #choicesDisplayed option', () => {
+    beforeEach(() => {
+      console.error = jest.fn();
+    });
+
+    afterEach(() => {
+      cleanup();
+    });
+    it('should properly render with the choices displayed', async () => {
+      const customChoices = [100, 250, 500];
+      const element = createTestComponent({
+        choicesDisplayed: customChoices.join(','),
+      });
+      await flushPromises();
+
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          initialState: {
+            numberOfResults: 100,
+          },
+        }
+      );
+
+      const resultsPerPage = element.shadowRoot.querySelector(
+        selectors.resultsPerPage
+      );
+      expect(resultsPerPage).not.toBeNull();
+
+      const resultsPerPageOptions = Array.from(
+        element.shadowRoot.querySelectorAll(selectors.resultsPerPageOption)
+      );
+      expect(resultsPerPageOptions).toHaveLength(customChoices.length);
+
+      resultsPerPageOptions.forEach((option, index) => {
+        expect(option.number).toEqual(customChoices[index]);
```

Comment:
also checking for the selected value:
```suggestion
        expect(option.number).toEqual(customChoices[index]);
        expect(option.selected).toBe(
          customChoices[index] === customChoices[0]
        );
```

Reply:
I didn't check that because in order to influence which value is selected, it has to go through headless.

Essentially, we have to mock the Headless state like this:

```
storeState = {
        resultsPerPage: {
          numberOfResults: 100,
        }
      };
```

In order to test that the `100` is actually the correct value selected.
It goes outside of the scope of the test that checks the option `choicesDisplayed` imo, as we already test that we're initializing the controller with the correct value in `numberOfResults` with this test:

```
expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledWith(
        exampleEngine,
        {
          initialState: {
            numberOfResults: 100,
          },
        }
      );
```

But I can add a test to make sure the Headless initial state is correctly rendered however.